auth.php

```diff
@@ -323,10 +338,88 @@ protected function is_ip_restriction_enabled() {
      * @return object User object.
      */
     protected function create_user(array $data) {
-        // TODO:
-        // 1. Validate user
-        // 2. Create user.
-        // 3. Throw exception if something went wrong.
+        global $DB, $CFG;
```

Comment:
As above - There's not really much to gain, this function does very little since we are only handling a tiny proportion of the fields on the user. It also imposes a lot of it's own validation which is either not relevant or would be a problem in this very specific use case.